crates/runtime/src/jit_int.rs

```diff
@@ -43,17 +46,25 @@ extern "C" fn __jit_debug_register_code() {
     }
 }
 
+lazy_static! {
+    pub static ref GDB_REGISTRATION: Mutex<u32> = Mutex::new(Default::default());
```

Comment:
Also as another note, this technically is still unsafe since we're not synchronized with anything else in the process that may be frobbing these symbols. Is there really no way we can rely on a gdb thing to do the synchronization for us?

Reply:
The gdb only known about `__jit_debug_descriptor` and sets breakpoint at `__jit_debug_register_code`. Locking access to __jit_debug_descriptor is the process chore -- gdb will traverse its data once process in the __jit_debug_register_code.